src/app/components/FundWalletModal/useFundWallet.ts

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react';
+import { useAppDispatch, useAppSelector } from 'app/hooks/reduxHooks';
+import { depositApi, useInitiateDepositMutation } from 'app/services/deposit/DepositService';
+import { encryptData } from 'app/helpers/encryptData';
+import { keyId, publicKey } from 'app/constants/constants';
+
+export const useFundWallet = (open: boolean) => {
+  const [deposit, setDeposit] = useState({
+    cvv: 0,
+    depositSize: 0,
+  });
+  const [error, setError] = useState('');
+
+  const dispatch = useAppDispatch();
+  const { id, data } = useAppSelector(state => state.creditCard.defaultCreditCard);
+  const [initiateDeposit, { isLoading, isSuccess, isError, data: depositData }] =
+    useInitiateDepositMutation();
+
+  const handleOnClick = async () => {
+    const data = await encryptData(publicKey, keyId, {
+      cvv: deposit.cvv,
+    });
+
+    await initiateDeposit({
+      keyId: keyId,
+      encryptedData: data.encryptedData,
+      cardId: id,
+      size: deposit.depositSize,
+    });
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setDeposit({ ...deposit, [name]: value });
+  };
+
+  useEffect(() => {
+    return () => {
```

Comment:
Why not adding this return on the other useEffect? Either way, the component is going to be unmounted, and it has nothing responding to open param.

Reply:
I will move this to only one effect, thanks